app/main/db/queries.py

```diff
@@ -62,81 +79,117 @@ def fuzzy_search(query_string):
         .distinct()
         .order_by(Body.Name, Series.Name)
     )
-    query_results = None
+    search_results = None
+    pages = 0
     try:
-        query_results = db.session.execute(query)
+        if total_records == 0:
+            query_results = db.session.execute(query)
+            total_records = len(query_results.all())
+
+        if total_records > 0:
+            json_result = paginate(
+                query,
+                record_count=total_records,
+                page=current_page,
+                per_page=per_page,
+            )
+            pages = json_result["pages"]
+            search_results = json_result["results"]
+
     except exc.SQLAlchemyError as e:
         print("Failed to return results from database with error : " + str(e))
 
-    if query_results is not None:
-        for r in query_results:
+    if search_results is not None:
+        for r in search_results:
             record = {
-                "transferring_body_id": r.body_id,
-                "transferring_body": r.transferring_body,
-                "series_id": r.series_id,
-                "series": r.series,
-                "consignment_reference": r.consignment_reference,
-                "file_name": r.file_name,
+                "transferring_body_id": r[0],
```

Comment:
I thought we decided that we'd stop parsing the query results list to a dictionary when we do the pagination

Reply:
yes, but default pagination supported by flasks-sqlalchemy run only on base query (i.e. direct model objects) and does not allow the results to paginate with joined queries, I could not find any solution which can help to avoid parsing query result